sdk/ml/azure-ai-ml/azure/ai/ml/operations/_index_operations.py

```diff
@@ -224,3 +235,249 @@ def cls(rest_indexes: Iterable[RestIndex]) -> List[Index]:
+        """
+        from azure.ai.ml.entities._indexes.entities.data_index import (
+            CitationRegex,
+            Data,
+            DataIndex,
+            Embedding,
+            IndexSource,
+            IndexStore,
+        )
+        from azure.ai.ml.entities._indexes.utils._open_ai_utils import build_open_ai_protocol, build_connection_id
+
+        if document_path_replacement_regex:
+            document_path_replacement_regex = json.loads(document_path_replacement_regex)
+
+        data_index = DataIndex(
+            name=name,
+            source=IndexSource(
+                input_data=Data(
+                    type="uri_folder",
+                    path=".",
+                ),
+                input_glob=input_glob,
+                chunk_size=tokens_per_chunk,
+                chunk_overlap=token_overlap_across_chunks,
+                citation_url=data_source_citation_url,
+                citation_url_replacement_regex=CitationRegex(
+                    match_pattern=document_path_replacement_regex["match_pattern"],  # type: ignore[index]
+                    replacement_pattern=document_path_replacement_regex["replacement_pattern"], # type: ignore[index]
+                )
+                if document_path_replacement_regex
+                else None,
+            ),
+            embedding=Embedding(
+                model=build_open_ai_protocol(
+                    model=embeddings_model_config.model_name,
+                    deployment=embeddings_model_config.deployment_name),
+                connection=build_connection_id(embeddings_model_config.connection_name, self._operation_scope),
+            ),
+            index=IndexStore(
+                type="acs",
+                connection=build_connection_id(index_config.ai_search_connection_id, self._operation_scope),
+                name=index_config.ai_search_index_name,
+            )
+            if index_config is not None
+            else IndexStore(type="faiss"),
+            # name is replaced with a unique value each time the job is run
+            path=f"azureml://datastores/workspaceblobstore/paths/indexes/{name}/{{name}}",
+        )
+
+        if isinstance(input_source, LocalSource):
+            data_index.source.input_data = Data(
+                type="uri_folder",
+                path=input_source.input_data.path,
+            )
+
+            return self.index_data(data_index=data_index, identity=input_source_credential)
+
+        if isinstance(input_source, str):
+            data_index.source.input_data = Data(
+                type="uri_folder",
+                path=input_source,
+            )
+
+            return self.index_data(data_index=data_index, identity=input_source_credential)
+
+        if isinstance(input_source, GitSource):
+            from azure.ai.ml.dsl import pipeline
+            from azure.ai.ml import MLClient
+            from azure.ai.ml.entities._indexes.data_index_func import index_data as index_data_func
```

Comment:
Please move as many of these imports to the top level as possible. (I suspect the MLClient import may introduce a circular import).